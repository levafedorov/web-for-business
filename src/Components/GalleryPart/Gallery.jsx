import React, { Component } from 'react';
import GalleryItem from './GalleryItem';
import Button from '../Buttons/Button';
import Positioner from '../Page/Positioner';
import PropTypes from 'prop-types';

export default class Gallery extends Component{
   

  state = {
     position: 0
  }
 
 
 leftHandler = (ev) => {
   const position = this.state.position;
   const max = this.props.photos.length - 1;
   if(position){
       this.setState({position: position - 1});
   }else{
    this.setState({position: max});
   }
 }


 rightHandler = (ev) => {
  const position = this.state.position;
  const max = this.props.photos.length - 1;
  if(position !== max){
      this.setState({position: position + 1});
  }else{
    this.setState({position: 0});
  }
}

  
 render(){

   const {position} = this.state;
  

   const {photos} = this.props;
   const total = photos.length;
   const itemWidth = (100 / total).toFixed(2);
   const screen =  {width: `${total * 100}%`,
                    transform: `translateX(-${itemWidth * position}%)`};
   

   return <div className="gallery" id="gallery">
       <div className="gallery-screen">
      <div className="screen" style={screen}>
      {photos.map((item, i) => <GalleryItem {...item} width={itemWidth} active={i === position} key={'photo-' + i}/>)}
      </div>
       </div>
       <div className="gallery-control">
       <Positioner>
       <Button class="rounded square-left" fn={this.leftHandler}/>
       <Button class="rounded square-right" fn={this.rightHandler}/>
       </Positioner>
       </div>
   </div>
 }
}

Gallery.defaultProps = {
 photos: [{photo:'', title:""}]
}