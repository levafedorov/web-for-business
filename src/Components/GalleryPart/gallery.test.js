import React from 'react';
import {shallow} from 'enzyme';
import Gallery from './Gallery';

const photos = 
[{photo: "photo", title: "photo"}, {photo: "photo", title: "photo"}, {photo: "photo", title: "photo"}];


describe('Gallery component', () => {
   let setUp = (props) => shallow(<Gallery {...props} />)
   let component;
   let instance;
   const context = {theme: 'default',  change: () => {}}

   beforeEach(() => {
     component = setUp({photos});
     instance =  component.instance();     
   });

   describe("check Gallery component with props and default props", () => {
       test('Check with props', () => {
        const component = setUp({photos});
        expect(component).toMatchSnapshot();
       });
       test('Check with defaultProps', () => {
        const component = setUp();
        expect(component).toMatchSnapshot();
       });
   });

   describe("Should width of GalleryItem change with amount of items", () => {
       test("width with 4 item", () => {
        const photosCopy = [...photos];
        photosCopy.push({photo: "photo", title: "photo"});
        const component = setUp({photos: photosCopy});
        expect(component).toMatchSnapshot();
       });
    });
    describe("check Handlers and position", () => {
        test("check leftHandler", () => {
          instance.leftHandler();
          expect(component).toMatchSnapshot();
        });
        test("check rightHandler", () => {
            instance.rightHandler();
            expect(component).toMatchSnapshot();
        });
     });

});