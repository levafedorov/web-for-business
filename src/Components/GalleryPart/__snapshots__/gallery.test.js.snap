// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component Should width of GalleryItem change with amount of items width with 4 item 1`] = `
<div
  className="gallery"
  id="gallery"
>
  <div
    className="gallery-screen"
  >
    <div
      className="screen"
      style={
        Object {
          "transform": "translateX(-0%)",
          "width": "400%",
        }
      }
    >
      <GalleryItem
        active={true}
        key="photo-0"
        photo="photo"
        title="photo"
        width="25.00"
      />
      <GalleryItem
        active={false}
        key="photo-1"
        photo="photo"
        title="photo"
        width="25.00"
      />
      <GalleryItem
        active={false}
        key="photo-2"
        photo="photo"
        title="photo"
        width="25.00"
      />
      <GalleryItem
        active={false}
        key="photo-3"
        photo="photo"
        title="photo"
        width="25.00"
      />
    </div>
  </div>
  <div
    className="gallery-control"
  >
    <Positioner>
      <Button
        class="rounded square-left"
        fn={[Function]}
      />
      <Button
        class="rounded square-right"
        fn={[Function]}
      />
    </Positioner>
  </div>
</div>
`;

exports[`Gallery component check Gallery component with props and default props Check with defaultProps 1`] = `
<div
  className="gallery"
  id="gallery"
>
  <div
    className="gallery-screen"
  >
    <div
      className="screen"
      style={
        Object {
          "transform": "translateX(-0%)",
          "width": "100%",
        }
      }
    >
      <GalleryItem
        active={true}
        key="photo-0"
        photo=""
        title=""
        width="100.00"
      />
    </div>
  </div>
  <div
    className="gallery-control"
  >
    <Positioner>
      <Button
        class="rounded square-left"
        fn={[Function]}
      />
      <Button
        class="rounded square-right"
        fn={[Function]}
      />
    </Positioner>
  </div>
</div>
`;

exports[`Gallery component check Gallery component with props and default props Check with props 1`] = `
<div
  className="gallery"
  id="gallery"
>
  <div
    className="gallery-screen"
  >
    <div
      className="screen"
      style={
        Object {
          "transform": "translateX(-0%)",
          "width": "300%",
        }
      }
    >
      <GalleryItem
        active={true}
        key="photo-0"
        photo="photo"
        title="photo"
        width="33.33"
      />
      <GalleryItem
        active={false}
        key="photo-1"
        photo="photo"
        title="photo"
        width="33.33"
      />
      <GalleryItem
        active={false}
        key="photo-2"
        photo="photo"
        title="photo"
        width="33.33"
      />
    </div>
  </div>
  <div
    className="gallery-control"
  >
    <Positioner>
      <Button
        class="rounded square-left"
        fn={[Function]}
      />
      <Button
        class="rounded square-right"
        fn={[Function]}
      />
    </Positioner>
  </div>
</div>
`;

exports[`Gallery component check Handlers and position check leftHandler 1`] = `
<div
  className="gallery"
  id="gallery"
>
  <div
    className="gallery-screen"
  >
    <div
      className="screen"
      style={
        Object {
          "transform": "translateX(-66.66%)",
          "width": "300%",
        }
      }
    >
      <GalleryItem
        active={false}
        key="photo-0"
        photo="photo"
        title="photo"
        width="33.33"
      />
      <GalleryItem
        active={false}
        key="photo-1"
        photo="photo"
        title="photo"
        width="33.33"
      />
      <GalleryItem
        active={true}
        key="photo-2"
        photo="photo"
        title="photo"
        width="33.33"
      />
    </div>
  </div>
  <div
    className="gallery-control"
  >
    <Positioner>
      <Button
        class="rounded square-left"
        fn={[Function]}
      />
      <Button
        class="rounded square-right"
        fn={[Function]}
      />
    </Positioner>
  </div>
</div>
`;

exports[`Gallery component check Handlers and position check rightHandler 1`] = `
<div
  className="gallery"
  id="gallery"
>
  <div
    className="gallery-screen"
  >
    <div
      className="screen"
      style={
        Object {
          "transform": "translateX(-33.33%)",
          "width": "300%",
        }
      }
    >
      <GalleryItem
        active={false}
        key="photo-0"
        photo="photo"
        title="photo"
        width="33.33"
      />
      <GalleryItem
        active={true}
        key="photo-1"
        photo="photo"
        title="photo"
        width="33.33"
      />
      <GalleryItem
        active={false}
        key="photo-2"
        photo="photo"
        title="photo"
        width="33.33"
      />
    </div>
  </div>
  <div
    className="gallery-control"
  >
    <Positioner>
      <Button
        class="rounded square-left"
        fn={[Function]}
      />
      <Button
        class="rounded square-right"
        fn={[Function]}
      />
    </Positioner>
  </div>
</div>
`;
